### PR TITLE
Sidecar in Hybrid Mode

### LINKED ISSUE #35111: Enable TLS termination on the envoy sidecar
(This is used to request new product features, please visit <https://discuss.istio.io> for questions on using Istio)

**Describe the feature request**
Enable TLS termination on the envoy sidecar.
Currently, the server side envoy proxy running as a sidecar can only allow mTLS for secure connection. In a regular Istio mesh deployment the onus of one way TLS termination for downstream requests is on the ingress gateway. Though this satisfies most of the use cases, for some use cases (like an API Gateway on mesh) the ingress gateway resource is not necessarily needed.

This feature request is to introduce an annotation "sidecar.istio.io/proxyMode: true" on a deployment that runs the Istio proxy on the pod in hybrid mode, i.e., it performs a function of both a router/gateway and a sidecar.

**Describe alternatives you've considered**
1. Create a custom EnvoyFilter configuration as described in [this](https://github.com/istio/istio/issues/21368) issue. The problem with this approach is that this could lead to maintenance overhead as EnvoyFilters does not guarantee backwards compatibility. 
2. Use the Ingress Gateway resource to offload the TLS termination. Going this route forces us to add an unnecessary network hop which can be avoided by this feature.

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**
Related Issue: https://github.com/istio/istio/issues/21368
RFC (WIP): https://docs.google.com/document/d/15Qhr7errbylXEzxxCK7ij_oUpn4E5SFU2uDdl_n2GIc/edit?usp=sharing

### PR DESCRIPTION
**Pull request by shriramsharma**
**Please provide a description of this PR:**
Fixes https://github.com/istio/istio/issues/35111
Extended sidecar to perform gateway functions in order to enable TLS from outside of mesh.
RFC: [Support hybrid sidecar mode](https://docs.google.com/document/d/15Qhr7errbylXEzxxCK7ij_oUpn4E5SFU2uDdl_n2GIc/edit#heading=h.xw1gqgyqs5b)

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [X] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/injection-template.yaml [hunk 1]
```diff
@@ -410,6 +412,11 @@ spec:
       name: lightstep-certs
       readOnly: true
     {{- end }}
+    {{ range .Values.global.proxy.hybrid.secretVolumes }}
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:415]: I don't think this is the right path forward, should be exposed over SDS

See also https://github.com/istio/istio/issues/32202; the main problem is auth

Comment (shriramsharma) [reply manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:415]: thanks @howardjohn , I'll fix this one

Also, should I move this PR to `master` as well, similar to the [API one](https://github.com/istio/api/pull/2165)?

Comment (howardjohn) [reply manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:415]: Yes, new code is only accepted on `master`. (sometimes) you can change it in the github UI by clicking at the top where it says release-1.12 without creating a new PR

Comment (shriramsharma) [reply manifests/charts/istio-control/istio-discovery/files/injection-template.yaml:415]: hey @howardjohn , there are just way too many conflicts after changing the branch. I'll just go ahead and create a new PR and link this PR for historical reference.

**Comment by shriramsharma** [comment]
Created a new PR on master https://github.com/istio/istio/pull/36210

**Pull request closed without merging**